tests/unit/s2n_tls13_handshake_test.c

```diff
@@ -333,23 +333,93 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_ecc_evp_generate_ephemeral_key(&conn->secure.client_ecc_evp_params[0]));
 
             const uint8_t psk_data[] = "test";
+            const uint8_t secret_data[] = "test secret data";
             for (size_t i = 0; i < S2N_TEST_PSK_COUNT; i++) {
                 struct s2n_psk *psk = NULL;
                 EXPECT_OK(s2n_array_pushback(&conn->psk_params.psk_list, (void**) &psk));
                 EXPECT_OK(s2n_psk_init(psk, S2N_PSK_TYPE_EXTERNAL));
                 EXPECT_SUCCESS(s2n_psk_set_identity(psk, psk_data, sizeof(psk_data)));
                 EXPECT_NOT_EQUAL(psk->identity.size, 0);
                 EXPECT_NOT_EQUAL(psk->identity.data, NULL);
+                EXPECT_SUCCESS(s2n_psk_set_secret(psk, secret_data, sizeof(secret_data)));
+                EXPECT_NOT_EQUAL(psk->secret.size, 0);
+                EXPECT_NOT_EQUAL(psk->secret.data, NULL);
             }
 
             EXPECT_NOT_EQUAL(conn->psk_params.psk_list.mem.allocated, 0);
             EXPECT_EQUAL(conn->psk_params.psk_list.len, S2N_TEST_PSK_COUNT);
+            EXPECT_NULL(conn->psk_params.chosen_psk);
 
             EXPECT_SUCCESS(s2n_tls13_handle_handshake_secrets(conn));
 
-            /* Verify PSKs are wiped */
-            EXPECT_EQUAL(conn->psk_params.psk_list.mem.allocated, 0);
-            EXPECT_EQUAL(conn->psk_params.psk_list.len, 0);
+            /* Verify secrets are wiped */
+            for (size_t i = 0; i < conn->psk_params.psk_list.len; i++) {
+                struct s2n_psk *psk = NULL;
+                EXPECT_OK(s2n_array_get(&conn->psk_params.psk_list, i, (void**)&psk));
+                EXPECT_NOT_NULL(psk->identity.data);
```

Comment:
EXPECT_NOT_EQUAL(psk->identity.size, 0) is probably a better test, if you're only going to do one blob attribute. A blob can point to data but be empty. (same on next test)